taichi/program/aot_module_builder.cpp

```diff
@@ -31,5 +31,31 @@ void AotModuleBuilder::add_kernel_template(const std::string &identifier,
   add_per_backend_tmpl(identifier, key, kernel);
 }
 
+bool AotModuleBuilder::all_fields_are_dense_in_container(
```

Comment:
I feel like this can have an even wider applicability, but the base AOT builder seems good enough :-)

Reply:
yup agreed! I'd just keep this here for now since we know it'll be useful for all aot backends for sure. Later we can move it to other places if needed :D